storage/filesystem/dotgit/dotgit.go

```diff
@@ -582,7 +582,7 @@ func (d *DotGit) hasIncomingObjects() bool {
 		directoryContents, err := d.fs.ReadDir(objectsPath)
 		if err == nil {
 			for _, file := range directoryContents {
-				if strings.HasPrefix(file.Name(), "incoming-") && file.IsDir() {
+				if strings.HasPrefix(file.Name(), "tmp_objdir-incoming-") && file.IsDir() {
```

Comment:
To be backwards compatible, we should handle the new prefix while keeping the previous.
Add a comment calling out what version (e.g. `v2.35.0`) of `git` made the change, so we can remove the previous prefix if we ever drop support below that version.